closed/src/java.base/share/classes/sun/security/ec/NativeECKeyPairGenerator.java

```diff
@@ -192,13 +211,30 @@ public void initialize(AlgorithmParameterSpec params, SecureRandom random)
         }
 
         // Not all known curves are supported by the native implementation
-        ECKeyPairGenerator.ensureCurveIsSupported(ecSpec);
+        try{
+            ECKeyPairGenerator.ensureCurveIsSupported(ecSpec);
+        } catch (InvalidAlgorithmParameterException iape) {
+            if (!(ECUtil.isBrainpoolP512r1(ecSpec))) {
+                throw iape;
+            }
+        }
+
         this.params = ecSpec;
 
         this.keySize = ecSpec.getCurve().getField().getFieldSize();
         this.random = random;
 
         this.curve = NativeECUtil.getCurveName(this.params);
+
+        /* Disabling OpenSSL usage on AIX due to performance regression observed. */
+        if (isAIX && !curve.equalsIgnoreCase("brainpoolP512r1")) {
```

Comment:
This should use `NativeECUtil.isBrainpoolP512r1()`:
```java
        if (OperatingSystem.isAix() && !NativeECUtil.isBrainpoolP512r1(ecSpec)) {
```
`isBrainpoolP512r1()` should be moved from `ECUtil` and made case-insensitive and more robust:
```java
    static boolean isBrainpoolP512r1(ECParameterSpec name) {
        NamedCurve curve = CurveDB.lookup(name);
        if (curve != null) {
            String[] nameAndAliases = curve.getNameAndAliases();
            for (String nameOrAlias : nameAndAliases) {
                if ("brainpoolP512r1".equalsIgnoreCase(nameOrAlias)) {
                    return true;
                }
            }
        }
        return false;
    }
```

Reply:
The `isBrainpoolP512r1` method was moved to `NativeECUtil` and removed from `ECUtil`.

The method was also enhanced to be more robust as suggested in that file.